atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -671,11 +686,23 @@ private void updateMetricOnAggregateAndHost(
                     String errorMsg = MessageFormatter.format(CONNECTION_FAILURE_MSG, numTries).getMessage();
                     throw (K) new TTransportException(((TTransportException) ex).getType(), errorMsg, ex);
                 } else {
-                    log.error("Tried to connect to cassandra {} times.", numTries, ex);
+                    log.error("Tried to connect to cassandra {} times.", SafeArg.of("numTries", numTries), ex);
                     throw (K) ex;
                 }
             } else {
-                log.warn("Error occurred talking to cassandra. Attempt {} of {}.", numTries, MAX_TRIES_TOTAL, ex);
+                // Only log the actual exception the first time
+                if (numTries == 1) {
+                    log.warn("Error occurred talking to cassandra. Attempt {} of {}. Exception message was: {} : {}",
+                            SafeArg.of("numTries", numTries),
+                            SafeArg.of("maxTotalTries", MAX_TRIES_TOTAL),
+                            SafeArg.of("exceptionClass", ex.getClass().getTypeName()),
+                            UnsafeArg.of("exceptionMessage", ex.getMessage()));
+                } else {
+                    log.warn("Error occurred talking to cassandra. Attempt {} of {}.",
+                            SafeArg.of("numTries", numTries),
+                            SafeArg.of("maxTotalTries", MAX_TRIES_TOTAL),
+                            ex);
```

Comment:
shouldn't the exception be the last arg in the if block, not else? I think we could actually do the following to cover all cases so that we include the exception class & message all the time, but only include stacktrace on first occurrence.

```
            log.warn("Error occurred talking to cassandra. Attempt {} of {}. Exception message was: {} : {}",
                    SafeArg.of("numTries", numTries),
                    SafeArg.of("maxTotalTries", MAX_TRIES_TOTAL),
                    SafeArg.of("exceptionClass", ex.getClass().getName()),
                    UnsafeArg.of("exceptionMessage", ex.getMessage()),
                    (numTries == 1) ? ex : null);
```

Reply:
....d'oh, I was right the first time!

Reply:
@gsheasby isn’t your logic here backwards if you’re trying to only log the exception on first failed attempt? I am pretty sure the whole of/else should be replaced with my comment above

Reply:
Looks correct on develop. We don't need to log exceptionClass or message on the first try, since we log the whole exception.